lib/speedbump_test.go

```diff
@@ -140,7 +140,20 @@ func TestSpeedbumpWithEchoServer(t *testing.T) {
 	assert.Nil(t, err)
 
 	tcpAddr, _ := net.ResolveTCPAddr("tcp", "localhost:8000")
-	conn, _ := net.DialTCP("tcp", nil, tcpAddr)
+
+	var conn *net.TCPConn
+
+	// Wait for the speedbump instance to start listening
+	// since it is started in a separate goroutine, we don't know
+	// if it has already started listening by this point
+	for {
+		conn, err = net.DialTCP("tcp", nil, tcpAddr)
+		if err != nil {
+			time.Sleep(10 * time.Millisecond)
+		} else {
+			break
+		}
+	}
```

Comment:
This test bug was causing the CI pipeline to fail every once in a while. Thanks for fixing it.

On a side note, it shows that the public API could be a bit more intuitive (i.e. `Start()` that returns immediately after the proxy starts listening). Such changes will likely be introduced in v1 due to their breaking nature.